saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2AuthenticationRequestContext.java

```diff
@@ -87,6 +98,32 @@ public String getRelayState() {
 		return this.relayState;
 	}
 
+	/**
+	 * Returns the desired ({@code RequestedAuthnContext} that this SP wishes to set in
+	 * the request.This property populates the {@code AuthNRequest.RequestedAuthnContext}
+	 * XML attribute.
+	 * @return the AuthnContextClassRef value
+	 */
+	public RequestedAuthnContext getRequestAuthnContext() {
```

Comment:
We don't want any OpenSAML in the public API. 

Instead, I'd recommend changing this to a collection to `String` values and renaming the method to `getRequestAuthnContextRefs`.

Reply:
Great point 👍 . I was also thinking of having it as a List of String, however, one challenge is we won't be able to set the `Comparison` value [AuthnContextComparisonTypeEnumeration](https://github.com/apigee/java-opensaml2/blob/master/src/main/java/org/opensaml/saml2/core/AuthnContextComparisonTypeEnumeration.java#L25). Shall we set it to `exact` as default?

Reply:
Yes, the spec indicates that "exact" is the default.